src/utils/push.js

```diff
@@ -1,72 +1,126 @@
-  const currentScreen = navigationRef.current?.getCurrentRoute()?.name;
-  const isSendbirdNotification = Boolean(remoteMessage.data.sendbird);
-  if (!isSendbirdNotification) return;
+  startOnForeground() {
+    const unsubscribes = [];
 
-  const channelId = await notifee.createChannel({
-    id: 'SendbirdNotificationChannel',
-    name: 'Sendbird RN Notification Sample',
-  });
+    if (Platform.OS === 'android') {
+      messaging().onMessage(async remoteMessage => {
+        if (isSendbirdNotification(remoteMessage.data)) {
+          const sendbird = parseSendbirdNotification(remoteMessage.data);
+          const currentScreen = navigationRef.current?.getCurrentRoute()?.name;
 
-  if (remoteMessage && remoteMessage.data) {
-    let pushActionId = 'SendbirdNotification-';
-    const sendbirdData = JSON.parse(remoteMessage.data.sendbird);
+          if (
+            currentScreen === 'Notifications' &&
+            sendbird.channel.channel_url === store.getState().sendbird.feedChannel._url
+          ) {
+            dispatch(updateHasNewNotifications(true));
+          } else {
+            dispatch(refreshCollection());
+            await this.displayNotification(remoteMessage, sendbird);
+          }
+        }
+      });
+      unsubscribes.push(
+        notifee.onForegroundEvent(async ({type, detail}) => {
+          if (type === EventType.PRESS && detail.notification && isSendbirdNotification(detail.notification.data)) {
+            const payload = parseSendbirdNotification(detail.notification.data);
+          }
+        }),
+      );
+    }
 
-    if (sendbirdData.channel.channel_url === FEED_CHANNEL_URL) {
-      pushActionId += sendbirdData.message_id;
+    if (Platform.OS === 'ios') {
+      PushNotificationIOS.addEventListener('notification', async notification => {
+        const data = notification.getData();
+        if (isSendbirdNotification(data)) {
+          const sendbird = parseSendbirdNotification(data);
+          const currentScreen = navigationRef.current?.getCurrentRoute()?.name;
+          if (
+            currentScreen === 'Notifications' &&
+            sendbird.channel.channel_url === store.getState().sendbird.feedChannel._url
+          ) {
+            dispatch(updateHasNewNotifications(true));
+          } else {
+            dispatch(refreshCollection());
+            this.displayNotification(notification, sendbird);
+          }
 
-      if (currentScreen !== 'Notifications') {
-        await notifee.displayNotification({
-          title: 'Sendbird RN Sample',
-          body: remoteMessage.data.message,
-          android: {
-            channelId,
-            pressAction: {
-              id: pushActionId,
-              launchActivity: 'default',
-            },
-          },
-        });
-        store.dispatch(refreshCollection());
-      } else {
-        store.dispatch(updateHasNewNotifications(true));
-      }
+          if (data.userInteraction === 1) {
+          }
```

Comment:
It seems not doing anything.

Reply:
If specifying the part that handles the notification when it's clicked, it would be a good idea to leave comments or logs!